snap/local/postinst.d/10_configure_auto_login

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+set -e
+
+installer_conf=/tmp/ubuntu_desktop_installer.conf
+
+if ! source $installer_conf 2>/dev/null || [ -z "$AutoLoginUser" ]; then
+    exit
+fi
+
+USER=$AutoLoginUser
+ROOT=/target
+chroot=chroot
+BACKUP=
+
+if [ -d "$ROOT/etc/gdm3" ]; then
+    # Configure GDM autologin
+    GDMCustomFile=$ROOT/etc/gdm3/custom.conf
+    if [ -e "$GDMCustomFile" ] && [ "$BACKUP" ]; then
+        cp "$GDMCustomFile" "${GDMCustomFile}$BACKUP"
+    fi
+    AutologinParameters="AutomaticLoginEnable=true\n\
+AutomaticLogin=$USER\n"
+
+    # Prevent from updating if parameters already present (persistent usb key)
+    if ! `grep -qs "AutomaticLogin=$USER" $GDMCustomFile` ; then
+        if [ -e "$GDMCustomFile" ]; then
+            sed -i '/\(Automatic\)Login/d' $GDMCustomFile
+        fi
+        if ! `grep -qs '\[daemon\]' $GDMCustomFile` ; then
+            echo '[daemon]' >> $GDMCustomFile
+        fi
+        sed -i "s/\[daemon\]/\[daemon\]\n$AutologinParameters/" $GDMCustomFile
+    fi
+fi
+
+if $chroot $ROOT [ -f /etc/kde4/kdm/kdmrc ]; then
+    # Configure KDM autologin
+    $log $chroot $ROOT sed -i$BACKUP -r \
```

Comment:
we don't have ubiquity's log-output tool but I left "$log" here regardless. it seems harmless to let it evaluate to empty?

Reply:
it isn't a functional issue but it's a bit crufty and at some point in the futur someone might wonder why it's there so I would be in favor of having them removed

Do you want to keep them to make easier to bring updated versions from that file from Debian? If so it seems they should be easier to filter out with a sed, maybe we can just document that at the top of the file?

Also please add a reference to the origin of the file, I don't expect the original to really see changes but it might help to ensure we keep them in sync if ever needed